### PR TITLE
Add GetStringUTFLengthAsLong for Java 24+

### PR DESCRIPTION
**Pull request by keithc-ca**
Also add missing JNI check wrappers for `GetModule` and `IsVirtualThread`.

### CODE REVIEW

File: runtime/j9vm31/jnimisc.cpp [hunk 1]
```diff
@@ -173,12 +173,26 @@ GetStringUTFLength(JNIEnv *env, jstring string)
...
+#if JAVA_SPEC_VERSION >= 24
+jlong JNICALL
+GetStringUTFLengthAsLong(JNIEnv *env, jstring string)
+{
+	const jint NUM_ARGS = 2;
+	J9_CEL4RO64_ArgType argTypes[NUM_ARGS] = { CEL4RO64_type_JNIEnv64, CEL4RO64_type_jstring };
+	uint64_t argValues[NUM_ARGS] = { JNIENV64_FROM_JNIENV31(env), string };
+	jlong returnValue = 0;
+	FUNCTION_DESCRIPTOR_FROM_JNIENV31(env, GetStringUTFLengthAsLong);
+	j9_cel4ro64_call_function(functionDescriptor, argTypes, argValues, NUM_ARGS, CEL4RO64_type_jboolean, &returnValue);
```
Comment (pshipton) [inline runtime/j9vm31/jnimisc.cpp:191]: This should be CEL4RO64_type_jlong.

File: runtime/jnichk/jnicwrappers.c [hunk 2]
```diff
@@ -4091,6 +4090,57 @@ checkGetObjectRefType(JNIEnv *env, jobject obj)
...
+checkIsVirtualThread(JNIEnv *env, jobject obj)
+{
+	jboolean actualResult = JNI_FALSE;
+	J9JniCheckLocalRefState refTracking;
+	static const U_32 argDescriptor[] = { JNIC_JOBJECT, 0 };
+	static const char function[] = "IsVirtualThread";
+
+	jniCheckArgs(function, 0, CRITICAL_WARN, &refTracking, argDescriptor, env, obj);
+	actualResult = globalJavaVM->EsJNIFunctions->IsVirtualThread(env, obj);
+	jniCheckFlushJNICache(env);
```
Comment (pshipton) [inline runtime/jnichk/jnicwrappers.c:4121]: Shouldn't this call jniCheckLocalRefTracking()?

Comment (keithc-ca) [reply runtime/jnichk/jnicwrappers.c:4121]: I'll add a call here and in `checkGetModule()`, but it's not clear to me that they're necessary for functions which can't return a reference (e.g. I don't think they're necessary in `checkGetStringUTFLength()` or `checkGetStringUTFLengthAsLong()`).

Comment (keithc-ca) [reply runtime/jnichk/jnicwrappers.c:4121]: I created https://github.com/eclipse-openj9/openj9/issues/20478 to follow up on uses of `jniCheckLocalRefTracking()`.

**Comment by keithc-ca** [comment]
Feedback addressed in ba6c33d0be2a6acb4ece0474b4a3a88ac714b82f.

**Comment by pshipton** [comment]
jenkins test sanity.functional win jdk21,jdknext

**Review by pshipton — APPROVED** [review]

**Pull request merged**